### PR TITLE
fix(apigee): Added in-place update of `google_apigee_environment` resource

### PR DESCRIPTION
**Pull request by entertvl**
- fix https://github.com/hashicorp/terraform-provider-google/issues/19820

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:bug
apigee: added in-place update of `google_apigee_environment` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/apigee/resource_apigee_environment_update_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccApigeeEnvironment_apigeeEnvironmentUpdateTest(t *testing.T) {
```
Comment (hao-nan-li) [inline mmv1/third_party/terraform/services/apigee/resource_apigee_environment_update_test.go:12]: What exactly is this test updating? Can you make this a 2 steps test to show the update?

Comment (entertvl) [reply mmv1/third_party/terraform/services/apigee/resource_apigee_environment_update_test.go:12]: Do you mean splitting `resource.TestStep` within one test case, or separating it into a different `func`?
Just wanted to confirm, thanks!

Comment (hao-nan-li) [reply mmv1/third_party/terraform/services/apigee/resource_apigee_environment_update_test.go:12]: Looks like I missed `testAccApigeeEnvironment_apigeeEnvironmentBasicTestExample` in the test first step. Is there any particular reason to skip the vcr test here?

Comment (entertvl) [reply mmv1/third_party/terraform/services/apigee/resource_apigee_environment_update_test.go:12]: To be honest, I don’t fully understand the exact reason for skipping VCR test.
In the past [PR](https://github.com/GoogleCloudPlatform/magic-modules/pull/11479), we had a same situation.

I’ll remove the VCR skip for now and see if it works.

File: mmv1/products/apigee/Environment.yaml [hunk 2]
```diff
@@ -114,13 +114,13 @@ properties:
     description: |
       Display name of the environment.
     required: false
-    immutable: true
+    immutable: false
```
Comment (hao-nan-li) [inline mmv1/products/apigee/Environment.yaml:117]: I noticed that this resource has multiple fields with `immutable` set to true. Is this PR change intended to make these 2 fields updatable, or the entire resource updatable?

Comment (entertvl) [reply mmv1/products/apigee/Environment.yaml:117]: This change makes only `DisplayName` and `Description`.

Although the following issue refers only to Description, DisplayName is also included as updatable:
https://github.com/hashicorp/terraform-provider-google/issues/19820

**Review by hao-nan-li — APPROVED** [review]

**Pull request merged**